apm-sniffer/apm-agent-core/pom.xml

```diff
@@ -104,6 +104,12 @@
             <version>${ststem-rules.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.openjdk.jmh</groupId>
+            <artifactId>jmh-generator-annprocess</artifactId>
+            <version>${jmh.version}</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
You are adding this for agent core only, so the datacarrier benchmark is still not runnable. I don't think this is the right way.